src/actions/preferences.js

```diff
@@ -1,43 +1,30 @@
 // Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
 // See License.txt for license information.
 
-import {batchActions} from 'redux-batched-actions';
-
 import Client from 'client';
 import {Preferences, PreferencesTypes} from 'constants';
 import {getMyPreferences as getMyPreferencesSelector} from 'selectors/entities/preferences';
 import {getCurrentTeamId} from 'selectors/entities/teams';
 import {getCurrentUserId} from 'selectors/entities/users';
 import {getPreferenceKey} from 'utils/preference_utils';
 
-import {bindClientFunc, forceLogoutIfNecessary} from './helpers';
-import {getLogErrorAction} from './errors';
+import {bindClientFunc} from './helpers';
 import {getProfilesByIds, getProfilesInChannel} from './users';
 
 export function deletePreferences(preferences) {
-    return async (dispatch, getState) => {
-        dispatch({type: PreferencesTypes.DELETE_PREFERENCES_REQUEST}, getState);
-
-        try {
-            await Client.deletePreferences(preferences);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch);
-            dispatch(batchActions([
-                {type: PreferencesTypes.DELETE_PREFERENCES_FAILURE, error},
-                getLogErrorAction(error)
-            ]), getState);
-            return;
-        }
-
-        dispatch(batchActions([
-            {
-                type: PreferencesTypes.DELETED_PREFERENCES,
-                data: preferences
-            },
-            {
-                type: PreferencesTypes.DELETE_PREFERENCES_SUCCESS
+    return async (dispatch) => {
+        dispatch({
+            type: PreferencesTypes.DELETE_PREFERENCES,
+            data: preferences,
+            meta: {
+                offline: {
+                    effect: () => Client.deletePreferences(preferences),
```

Comment:
another worry that I have, what if this action is retrying after a while and the session has expired? can we stop retrying if the response has status 401?